tests/spec_tests/query_tests/distinct.yml

```diff
@@ -0,0 +1,154 @@
+catalog_data:
+  db:
+    foo:
+      - { _id: 1, a: 1, b: 2 }
+      - { _id: 2, a: 1, b: 2 }
+      - { _id: 3, a: 2, b: 2 }
+      - { _id: 4, b: 2, a: 1 }
+      - { _id: 5, a: 2, b: 2, c: true, d: "a", e: null }
+      - { _id: 6, a: 2, b: 2, c: true, d: "a", e: null }
+
+    bar:
+      - { _id: 1, a: 1, b: 2 }
+      - { _id: 2, a: 2, b: 2 }
+      - { _id: 3, a: 2, b: 2 }
+      - { _id: 4, a: 2, b: 4 }
+
+    documents:
+      - { _id: 1, doc: { a: 1, b: 2 } }
+      - { _id: 2, doc: { a: 1, b: 2 } }
+      - { _id: 3, doc: { a: 2, b: 2 } }
+      - { _id: 4, doc: { a: 2, b: 3 } }
+      - { _id: 5, doc: { b: 2, a: 1 } }
```

Comment:
I think the behavior should definitely work as expected, but would it be worth adding a doc (and an array below) that are subsets of another? eg `- {_id: 6, doc: {a: 1, b:2, c: 3 } }`? To verify we dont get caught by only checking the overlapping subset of keys.

Reply:
Sure, I added the doc and array with the updated results.